app/main/views/suppliers.py

```diff
@@ -580,12 +580,22 @@ def duns_number():
                 'duns_errors': ",".join(form.duns_number.errors)})
 
     errors = get_errors_from_wtform(form)
+    error_title = None
+    error_description = None
+    if errors.get("duns_number", {}).get("message", None) == 'DUNS number already used':
+        error_title = "A supplier account already exists with that DUNS number"
+        support_email_address = current_app.config['SUPPORT_EMAIL_ADDRESS']
+        error_description = Markup(
+            'If you no longer have your account details, or if you think this may be an error, '
+            f'email <a class="govuk-link" href="mailto:{support_email_address}?subject=DUNS%20number%20question" '
+            f'title="Please contact {support_email_address}">{support_email_address}</a>')
```

Comment:
What was the reasoning behind moving this out of the template?

Reply:
We removed all of the other custom error logic / messages from the templates (eg. [app/templates/frameworks/updates.html](https://github.com/alphagov/digitalmarketplace-supplier-frontend/pull/1340/files#diff-2dc1079cb1ccc1cf010d712b6ac4d4e01d9c0872852307d69d4a3dd54fabfb57L30)) when switching out the component. Everywhere else has the logic in the python side, so we moved this out of the template so it's in the same place for all pages.